src/App.tsx

```diff
@@ -1,24 +1,196 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoItem } from './components/TodoItem';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterOptions } from './types/FilterOptions';
+import { getFilteredTodos } from './utils/filterTodos';
+import { ErrorMessages } from './types/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 6345;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages>(
+    ErrorMessages.NoError,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<FilterOptions>(
+    FilterOptions.All,
+  );
+  const [loadingTodoIds, setLoadingTodoIds] = useState([0]);
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      setErrorMessage(ErrorMessages.NoError);
+      await deleteTodo(todoId);
+
+      setTodos((prevTodos: Todo[]): Todo[] => {
+        return prevTodos.filter(item => item.id !== todoId);
+      });
+    } catch (error) {
+      setErrorMessage(ErrorMessages.OnDelete);
+    }
+  };
+
+  const handleDeleteTodo = useCallback(async (todoId: number) => {
+    setLoadingTodoIds(prevIds => ([...prevIds, todoId]));
+    await removeTodo(todoId);
+    setLoadingTodoIds(prevIds => prevIds.filter(prevId => prevId !== todoId));
+  }, []);
+
+  const deleteCompletedTodos = useCallback(() => {
+    const completedTodosToDelete = getFilteredTodos(
+      todos,
+      FilterOptions.Completed,
+    );
+
+    completedTodosToDelete.forEach(todo => handleDeleteTodo(todo.id));
+  }, [todos]);
+
+  const visibleTodos = getFilteredTodos(todos, filterType);
+
+  const isSomeTodosCompleted: boolean = todos.some(item => item.completed);
+
+  const isAllTodosCompleted: boolean = todos.every(item => item.completed);
+
+  const handleTogglingTodo = useCallback(async (
+    todoId: number,
+    todoStatus: boolean,
+  ) => {
+    setLoadingTodoIds(prevIds => ([...prevIds, todoId]));
+    try {
+      await updateTodoStatus(todoId, todoStatus);
+      setErrorMessage(ErrorMessages.NoError);
+
+      setTodos((prevTodos: Todo[]): Todo[] => {
+        return prevTodos.map((item: Todo) => {
+          if (item.id === todoId) {
+            item.completed = todoStatus; // eslint-disable-line no-param-reassign
+          }
```

Comment:
can we find another way without skip linter rules, disable linter it's very bad practice & used only in extra rare cases